spring-cloud-contract-verifier/src/main/groovy/org/springframework/cloud/contract/verifier/util/ContractVerifierDslConverter.groovy

```diff
@@ -121,4 +111,34 @@ class ContractVerifierDslConverter {
 		}
 		return [object] as Collection<Contract>
 	}
+
+	private static Collection<Contract> listOfContracts(File file, Object object) {
+		if (object instanceof Collection) {
+			return withName(file, object as Collection<Contract>)
+		} else if (!object instanceof Contract) {
+			throw new DslParseException("Contract is not returning a Contract or list of Contracts")
+		}
+		return withName(file, [object] as Collection<Contract>)
+	}
+
+	private static Collection<Contract> withName(File file, Collection<Contract> contracts) {
+		int counter = 0
+		return contracts.collect {
+			if (hasName(it) && !relatedToScenarios(file, it)) {
+				String tillExtension = file.name.substring(0, file.name.lastIndexOf("."))
+				it.name(tillExtension + (counter > 0 || contracts.size() > 1 ? "_" + counter : ""))
+			}
+			counter++
+			return it
+		}
+	}
+
+	private static boolean hasName(Contract it) {
+		return it != null && !it.name
```

Comment:
it seems either the logic or the name should be modified cause in reality we check for non-existent name